src/core_plugins/kibana/public/dashboard/selectors/index.js

```diff
@@ -0,0 +1,32 @@
+import {
+  getEmbeddableTitle as getEmbeddableTitleFromEmbeddables,
+  getEmbeddableEditUrl as getEmbeddableEditUrlFromEmbeddables,
+  getEmbeddableError as getEmbeddableErrorFromEmbeddables,
+  getEmbeddable as getEmbeddableFromEmbeddables,
+} from './embeddables';
+
+import {
+  getPanel as getPanelFromPanels,
+  getPanelType as getPanelTypeFromPanels
+} from './panels';
+
+import {
+  getViewMode as getViewModeFromView,
+  getFullScreenMode as getFullScreenModeFromView,
+  getMaximizedPanelId as getMaximizedPanelIdFromView
+} from './view';
+
+export const getPanels = state => state.panels;
+export const getPanel = (state, panelId) => getPanelFromPanels(getPanels(state), panelId);
+export const getPanelType = (state, panelId) => getPanelTypeFromPanels(getPanels(state), panelId);
+
+export const getEmbeddables = state => state.embeddables;
+export const getEmbeddable = (state, panelId) => getEmbeddableFromEmbeddables(getEmbeddables(state), panelId);
+export const getEmbeddableError = (state, panelId) => getEmbeddableErrorFromEmbeddables(getEmbeddables(state), panelId);
```

Comment:
What's with all the indirection here? It's kind of tricky to follow exactly what is happening. It's reading state and passing the output as state into another selector, and that selector is using another selector in another file.

Reply:
Lets sync on this (perhaps with @chrisronline to represent the management team as well) so we can get on the same page.  I thought I was following the convention but perhaps not! :)